### PR TITLE
Adding title to nearby based on geocoding

### PR DESCRIPTION
**Pull request by coreyfloyd**
Add geocoded title and serialize with the section

### CODE REVIEW

**Comment by montehurd** [comment]
screen shot 2016-02-16 at 12 09 45 pm

**Comment by montehurd** [comment]
@coreyfloyd can you have a peek at https://phabricator.wikimedia.org/T126883 and see if it rings a bell? In the screenshot above notice all my distances are -3 feet too...

**Comment by coreyfloyd** [comment]
@montehurd not sure - didn't touch any of that code in this pr, we have a bug for it so can we can prioritize it accordingly.

**Comment by montehurd** [comment]
@coreyfloyd k. w/o location it's hard to test the new reverse geo bits - will try on device...

**Comment by coreyfloyd** [comment]
@bgerstle @montehurd anything feedback here?

**Comment by bgerstle** [comment]
:+1: works for me in sim:

image

and on device (redacting my address):

image

File: Wikipedia/Code/WMFExploreSectionSchema.m [hunk 1]
```diff
@@ -269,23 +270,43 @@ - (void)insertNearbySectionWithLocationIfNeeded:(CLLocation*)location {
...
     }];
+}
 
-    [sections wmf_safeAddObject:[self nearbySectionWithLocation:location]];
+typedef void (^ WMFGeocodeCompletionHandler)(CLPlacemark* __nullable placemark);
 
-    [self updateSections:sections];
+- (void)reverseGeocodeLocation:(CLLocation*)location completionHandler:(nonnull WMFGeocodeCompletionHandler)completionHandler {
+    CLGeocoder* gc = [[CLGeocoder alloc] init];
+    [gc reverseGeocodeLocation:location completionHandler:^(NSArray < CLPlacemark* > * _Nullable placemarks, NSError* _Nullable error) {
```
Comment (bgerstle) [inline Wikipedia/Code/WMFExploreSectionSchema.m]: FYI, PromiseKit has a bunch of [CoreLocation categories](https://github.com/mxcl/PromiseKit/blob/master/Categories/CoreLocation/CLGeocoder%2BPromise.swift) that we can include via subspec if desired.

File: Wikipedia/Code/WMFExploreSection.h [hunk 2]
```diff
@@ -87,6 +87,12 @@ typedef NS_ENUM (NSUInteger, WMFExploreSectionType){
 @property (nonatomic, strong, readonly) CLLocation* location;
 
 /**
+ *  The placemark associated with the section, if any.
+ *
+ */
+@property (nonatomic, strong, readonly) CLPlacemark* placemark;
```
Comment (bgerstle) [inline Wikipedia/Code/WMFExploreSection.h]: at first, i was wondering "wait, how are we writing this to disk" then I realized `CLPlacemark` already conforms to `NSSecureCoding` :+1:

**Pull request merged**